state/mapstate/migrate.go

```diff
@@ -89,11 +123,11 @@ func (st *MapState) migrateFrom(version int, snap []byte) error {
 	for {
 		next = m.next()
 		if next == nil {
-			mst2, ok := m.(*mapStateV2)
+			mst3, ok := m.(*mapStateV3)
 			if !ok {
 				return errors.New("migration ended prematurely")
 			}
-			finalCopy(st, mst2)
+			finalCopy(st, mst3)
```

Comment:
I'm really interested to hear your thoughts about the current approach to updating the state.  Was it straightforward to figure out what had to be done for this specific instance? Were the reference directions at the top helpful?  Did you have any ideas for improvements to the framework while implementing this change?

Reply:
It was straightfoward to copy paste the stuff and move on. I don't know if this can be done more elegantly, but works for me for the moment!